### ISSUE TITLE
Horizontal Scrollbar Not Removed Fully From DOM When Hidden

### ISSUE THREAD

**Issue by bjhobson12** [ag-grid/ag-grid#5924]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

**Providing a Reproducible Scenario**
- Creation of an AG grid with minimal cell padding allows for the last row rendered to be non-selectable. The horizontal scrollbar component partially takes up the space of the last row, making selection indirectly disabled.

**Current behavior**
- When an ag grid is rendered in react using ag-grid-react and the rendered columns do not fully take up the width of the grid (making the horizontal scrollbar hidden), the final row is not fully selectable, that is, the horizontal scrollbar partially overlaps the final row making it unclickable. When the browser width changes, the element with this class (ag-body-horizontal-scroll) responds once the with of the ag grid becomes less than the total column aggregated width; this renders the horizontal scroll bar. However, when the browser is expanded so this scroll bar is not visible, it still occupies the space it does when the element is visible. 

**Expected behavior**
- The expected behavior is that the horizontal scroll bar should not continue to occupy that space in the ag grid layout. This issue has been remediated in the private application I maintain by adding a class to the scrollbar element and listening with a mutation observer for changes to the element's inline style; once the style changes and the element is given height (making it visible on the page) the class is removed, but in all other scenarios it is applied to the element.

The class simply applies overflow: hidden to the element and it also selects children of that class.

```
.over-flow-hide, .over-flow-hide * {
    overflow: hidden;
}
```

This is when the scrollbar should be hidden
```
<div class="ag-body-horizontal-scroll over-flow-hide" aria-hidden="true" style="height: 17px; max-height: 17px; min-height: 17px;">
    <div class="ag-horizontal-left-spacer ag-scroller-corner" ref="eLeftSpacer" style="width: 0px; max-width: 0px; min-width: 0px;"></div>
    <div class="ag-body-horizontal-scroll-viewport" ref="eViewport" style="height: 17px; max-height: 17px; min-height: 17px;">
        <div class="ag-body-horizontal-scroll-container" ref="eContainer" style="height: 17px; max-height: 17px; min-height: 17px; width: 6525px;"></div>
    </div>
    <div class="ag-horizontal-right-spacer ag-scroller-corner" ref="eRightSpacer" style="width: 17px; max-width: 17px; min-width: 17px;"></div>
</div>
```

This is when the scrollbar should be visible
```
<div class="ag-body-horizontal-scroll" aria-hidden="true" style="height: 17px; max-height: 17px; min-height: 17px;">
    <div class="ag-horizontal-left-spacer ag-scroller-corner" ref="eLeftSpacer" style="width: 0px; max-width: 0px; min-width: 0px;"></div>
    <div class="ag-body-horizontal-scroll-viewport" ref="eViewport" style="height: 17px; max-height: 17px; min-height: 17px;">
        <div class="ag-body-horizontal-scroll-container" ref="eContainer" style="height: 17px; max-height: 17px; min-height: 17px; width: 6525px;"></div>
    </div>
    <div class="ag-horizontal-right-spacer ag-scroller-corner" ref="eRightSpacer" style="width: 17px; max-width: 17px; min-width: 17px;"></div>
</div>
```

**Please tell us about your environment:**
Running Intellij on linux 

**Version**
"ag-grid-community": "^25.3.0", "ag-grid-enterprise": "^25.3.0", "ag-grid-react": "^25.3.0"

**Browser:**
I am not allowed to share this information. This issue is found on chrome.

**Language:** [all | TypeScript X.X | ES6/7 | ES5]

**Comment by StephenCooper** [comment]
This sounds similar to the below issue which contains further information. 
https://github.com/ag-grid/ag-grid/issues/5534

We also included a fix in AG-7242 in v28.2 to prevent this issue happening too.

**Comment by bjhobson12** [comment]
Thanks Stephen. I’ll be sure to upgrade to v28. I appreciate your assistance.
